include/arch/arm/cortex_m/mpu/arm_core_mpu_dev.h

```diff
@@ -32,6 +32,9 @@ extern "C" {
  *
  */
 enum {
+#ifdef CONFIG_NOCACHE_MEMORY
+	NOCACHE_MEMORY_DATA_REGION,
```

Comment:
Is this going to be a "kernel" section only?

Reply:
I believe so, at least it's one I have done for the MPU permissions, and it matches the original case of doing DMA within the drivers.

Now this is an RFC because I believe we want something that can be available on other MPU or even other architectures. If there are some uses case that would also need to setup uncached regions in the user memory, we can still change that.